ports/stm32f4/common-hal/busio/SPI.c

```diff
@@ -23,37 +24,320 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+#include <stdbool.h>
 
 #include "shared-bindings/busio/SPI.h"
 #include "py/mperrno.h"
 #include "py/runtime.h"
+#include "stm32f4xx_hal.h"
 
+#include "shared-bindings/microcontroller/__init__.h"
 #include "boards/board.h"
+#include "supervisor/shared/translate.h"
 #include "common-hal/microcontroller/Pin.h"
 
+STATIC bool reserved_spi[6];
+
+void spi_reset(void) {
```

Comment:
You'll need to add a `never_reset` check here when you add external flash support otherwise the filesystem will stop working on first reset.

Reply:
Ah, that's what that's for. I guess I was thinking somehow that it used a different API. I'll implement that today.